### PR TITLE
Feature/4.2 view all books

### PR DESCRIPTION
**Pull request by Zokster9**
Add a book list with book cards. Add infinite scroll with pagination for the book list

### CODE REVIEW

File: library-valcon-internship/src/components/BookCard/BookCard.tsx [hunk 1]
```diff
@@ -0,0 +1,61 @@
+import placeholderBook from '../../assets/icons/placeholder-book.png'
+import Book from '../../models/Book'
+import './BookCard.css'
+
+interface BookCardProps {
+  book: Book
+}
+
+const BookCard = ({ book }: BookCardProps) => {
+  let descriptionLen = 0
```
Comment (aleksandarkrnjaic) [inline library-valcon-internship/src/components/BookCard/BookCard.tsx:10]: len???

File: library-valcon-internship/src/components/BookCard/BookCard.tsx [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+  let author = ''
+  if (book.Authors.length > 0) {
+    if (book.Authors.length > 1) {
+      author = `${book.Authors[0].FirstName} ${book.Authors[0].LastName}...`
+    } else {
+      author = `${book.Authors[0].FirstName} ${book.Authors[0].LastName}`
+    }
+  } else {
+    author = 'Unknown'
+  }
```
Comment (aleksandarkrnjaic) [inline library-valcon-internship/src/components/BookCard/BookCard.tsx:25]: You can always get all authors and then control the display with: https://caniuse.com/?search=line-clamp

File: library-valcon-internship/src/pages/HomePage/HomePage.tsx [hunk 3]
```diff
@@ -1,12 +1,49 @@
...
   return (
     <div className='homePage'>
-      <h3>No books currently available</h3>
+      {
+        books ?
+          (
+            <InfiniteScroll
+              dataLength={books.length}
+              next={handleNextPage}
+              hasMore={!isEmptyResponse}
```
Comment (aleksandarkrnjaic) [inline library-valcon-internship/src/pages/HomePage/HomePage.tsx:37]: How is has more false only if it's an empty response for previous request? That means we always have an extra request. Has more should be calculated based on total elements, current page and num of elements per page

Comment (Zokster9) [reply library-valcon-internship/src/pages/HomePage/HomePage.tsx:37]: The request does not return the total number of elements so I cannot calculate if there are no more books without an additional request.

**Pull request merged**